steward/src/cellars/aave_v2_stablecoin.rs

```diff
@@ -1,77 +1,182 @@
+            Ok(AaveV2StablecoinCellarCalls::Rebalance(call).encode())
         }
         Reinvest(params) => {
             let call = ReinvestCall {
                 min_assets_out: params.min_assets_out.into(),
             };
-            AaveV2StablecoinCellarCalls::Reinvest(call).encode()
+            Ok(AaveV2StablecoinCellarCalls::Reinvest(call).encode())
         }
-        ClaimAndUnstakeAmount(params) => {
-            let call = ClaimAndUnstakeWithAmountCall {
-                amount: params.amount.into(),
+        Sweep(params) => {
+            let token = match params.token.parse::<H160>() {
+                Ok(t) => t,
+                Err(_) => {
+                    return Err(sp_call_error(format!(
+                        "{}: Sweep 'token': argument is an invalid address: {}",
+                        LOG_PREFIX, params.token
+                    )))
+                }
             };
-            AaveV2StablecoinCellarCalls::ClaimAndUnstakeWithAmount(call).encode()
-        }
-        ClaimAndUnstake(_) => {
-            let call = ClaimAndUnstakeCall {};
-            AaveV2StablecoinCellarCalls::ClaimAndUnstake(call).encode()
-        }
-        Rebalance(params) => {
-            let call = RebalanceCall {
-                new_lending_token: params
-                    .address
-                    .parse::<H160>()
-                    .expect("failed to parse token address"),
-                min_new_lending_token_amount: params.min_new_lending_token_amount.into(),
+            let to = match params.to.parse::<H160>() {
+                Ok(t) => t,
+                Err(_) => {
+                    return Err(sp_call_error(format!(
+                        "{}: Sweep 'to': argument is an invalid address: {}",
+                        LOG_PREFIX, params.to
+                    )))
+                }
             };
-            AaveV2StablecoinCellarCalls::Rebalance(call).encode()
-        }
-        AccruePlatformFees(_) => {
-            let call = AccruePlatformFeesCall {};
-            AaveV2StablecoinCellarCalls::AccruePlatformFees(call).encode()
+            let call = SweepCall { token, to };
+            Ok(AaveV2StablecoinCellarCalls::Sweep(call).encode())
         }
         TransferFees(_) => {
             let call = TransferFeesCall {};
-            AaveV2StablecoinCellarCalls::TransferFees(call).encode()
+            Ok(AaveV2StablecoinCellarCalls::TransferFees(call).encode())
         }
-        SetInputToken(params) => {
-            let call = SetInputTokenCall {
-                token: params
-                    .address
-                    .parse::<H160>()
-                    .expect("failed to parse token address"),
-                is_approved: params.is_approved,
-            };
-            AaveV2StablecoinCellarCalls::SetInputToken(call).encode()
+    }
+}
+
+fn validate_route(results: Vec<Result<H160, &String>>) -> Result<(), Error> {
+    let mut bad_addresses = String::new();
```

Comment:
Did you want this to be a `Vec<String>`?

Reply:
No, lines 48-55 return bad addresses as the `&String`s for logging. Keeping it in this form seemed to be the cleanest way to validate with one log message and continue:

https://github.com/PeggyJV/steward/blob/8a2209a8cf456a62d2ce7d21de27541336c4a022/steward/src/cellars/aave_v2_stablecoin.rs#L48-L65

Reply:
No, renaming to `bad_addresses_string`